src/js/components/Box/StyledBox.js

```diff
@@ -67,32 +67,68 @@ const basisStyle = css`
 // https://stackoverflow.com/questions/36247140/why-doesnt-flex-item-shrink-past-content-size
 // we assume we are in the context of a Box going the other direction
 // TODO: revisit this
-const directionStyle = (direction, theme) => {
+const directionStyle = (directionWrapper, theme) => {
+  let direction;
+  let isResponsive;
+  let isReverse;
+
+  if (
+    typeof directionWrapper === 'string' ||
+    directionWrapper instanceof String
+  ) {
+    direction = directionWrapper;
+  } else {
+    // DirectionWrapper is object now.
+    direction = directionWrapper.direction;
+    isResponsive = directionWrapper.responsive;
+    isReverse = directionWrapper.reverse;
+  }
+
   const styles = [
     css`
       min-width: 0;
       min-height: 0;
       flex-direction: ${direction === 'row-responsive' ? 'row' : direction};
     `,
   ];
-  if (direction === 'row-responsive' && theme.box.responsiveBreakpoint) {
-    const breakpoint = getBreakpointStyle(
-      theme,
-      theme.box.responsiveBreakpoint,
-    );
-    if (breakpoint) {
-      styles.push(
-        breakpointStyle(
-          breakpoint,
-          `
+
+  const breakpoint = getBreakpointStyle(theme, theme.box.responsiveBreakpoint);
```

Comment:
why is this line being extracted outside of the 'if' (where it was placed before)?

Reply:
It was placed inside the if statement. Now we have two if's so don't wanted to duplicate it. Now added back to the if statement.